src/main/java/co/rsk/federate/signing/hsm/message/ReleaseCreationInformation.java

```diff
@@ -48,6 +58,32 @@ public ReleaseCreationInformation(
         this.pegoutCreationRskTxHash = pegoutCreationRskTxHash;
         this.pegoutBtcTx = pegoutBtcTx;
         this.pegoutConfirmationRskTxHash = pegoutConfirmationRskTxHash;
+
+        this.utxoOutpointValues = decodeUtxoOutpointValues(transactionReceipt);
+    }
+
+    private List<Coin> decodeUtxoOutpointValues(TransactionReceipt transactionReceipt) {
+        return transactionReceipt.getLogInfoList().stream()
+            .filter(this::isPegoutTransactionCreatedLog)
+            .findFirst()
+            .map(LogInfo::getData)
+            .map(this::decodePegoutTransactionEventData)
+            .map(UtxoUtils::decodeOutpointValues)
+            .orElse(Collections.emptyList());
+    }
+
+    private boolean isPegoutTransactionCreatedLog(LogInfo log) {
+        Function pegoutTransactionCreatedEvent = BridgeEvents.PEGOUT_TRANSACTION_CREATED.getEvent();
+        final byte[] pegoutTransactionCreatedSignatureTopic = pegoutTransactionCreatedEvent.encodeSignatureLong();
+
+        return !log.getTopics().isEmpty() && Arrays.equals(log.getTopics().get(0).getData(),
```

Comment:
Why assume it's always on index 0? There are other logs in the receipt, right? And there could be more in the future

Reply:
This is not for the logs, this is for the topics. The encoded signature (Keccak256 hash of the signature) will always be at index 0. The other indices for the same topic will be the `indexed` data of the event.

Reply:
To visualize it better, it looks like this:

logs = [
    LogInfo {
       topics: [
           [Keccak256 hash of the event signature]: hash,                   // update_collections(address)
       ]
    },
    LogInfo {
       topics: [
           [Keccak256 hash of the event signature]: hash,                  // release_requested(bytes32,bytes32,uint256)
            [First indexed event param data]: encoded_data,
            [Second indexed event param data]: encoded_data,
       ]
    },
   LogInfo {
       topics: [
           [Keccak256 hash of the event signature]: hash,                  // batch_pegout_created(bytes32,bytes)
            [First indexed event param data]: encoded_data,
       ]
    },
    LogInfo {
       topics: [
           [Keccak256 hash of the event signature]: hash,                // pegout_transaction_created(bytes32,bytes)
            [First indexed event param data]: encoded_data,
       ]
    }
}

Reply:
This zero index checks topics, not logs. We assume that the zero index topic always has the encoded signature of the event, so we check that the value matches the `pegoutTransactionCreatedEvent` signature to confirm this log belongs to `pegoutTransactionCreatedEvent`

Reply:
According to the API of logs and events, the zero index topic always has the signature of the event, so in fact, we are not assuming, we are just following the API.

Reply:
https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP146.md#topics

Reply:
True, my bad. Thanks!